themes/origins/assets/component-modal.js

```diff
@@ -94,9 +96,10 @@ class Modal extends HTMLElement {
     this.setState(!this.state);
   }
 
-  setState(hidden) {
-    this.state = hidden;
-    this.dataset.hidden = hidden;
+  setState(isVisible) {
```

Comment:
since the possible values are binary we can rename the method to setIsVisible to make it more readable
```suggestion
  setIsVisible(isVisible) {
```